File: exercises/smelly-tic-tac-toe/TypeScript/src/Board.ts
import {Tile} from "./Tile";

export class Board {
    private _plays: Tile[] = [];

    constructor() {
        for (let i = 0; i < 3; i++) {
            for (let j = 0; j < 3; j++) {
                const tile: Tile = {x: i, y: j, symbol: ' '};
                this._plays.push(tile);
            }
        }
    }

    public tileAt(x: number, y: number): Tile {
        return this._plays.find((t: Tile) => t.x == x && t.y == y)!;
    }

    public addTileAt(symbol: string, x: number, y: number): void {
        const tile: Tile = {x: x, y: y, symbol: symbol};

        this._plays.find((t: Tile) => t.x == x && t.y == y)!.symbol = symbol;
    }
}

File: exercises/smelly-tic-tac-toe/java/src/main/java/Board.java
import java.util.ArrayList;
import java.util.List;

public class Board {
    private List<Tile> plays = new ArrayList<>();

    public Board() {
        for (int i = 0; i < 3; i++) {
            for (int j = 0; j < 3; j++) {
                Tile tile = new Tile();
                tile.x = i;
                tile.y = j;
                tile.symbol = ' ';
                plays.add(tile);
            }
        }
    }

    public Tile tileAt(int x, int y) {
        for (Tile t : plays) {
            if (t.x == x && t.y == y) {
                return t;
            }
        }
        return null;
    }

    public void addTileAt(char symbol, int x, int y) {
        Tile newTile = new Tile();
        newTile.x = x;
        newTile.y = y;
        newTile.symbol = symbol;

        tileAt(x, y).symbol = symbol;
    }
}